### ISSUE TITLE
More detailed error messages

### ISSUE THREAD

**Issue by muziqaz** [FoldingAtHome/fah-client-bastet#273]
Not sure if this is fahclient side or web control side, but more detailed (less cryptic) error messages would help a lot in troubleshooting.

**Comment by jcoffland** [comment]
There's no easy way to make all error messages understandable for most users.  Many of the error messages are not meant to be understood by the end user because explaining many of the errors would be difficult if not impossible.  They are meant to be passed on via a bug report to the developer.

**Comment by muziqaz** [comment]
I recon couple of word hint would be helpful to us, more advanced users. Let's say instead of
```
10:58:34:I1:WU6:Uploading WU results
10:58:34:I1:OUT1506:> POST https://ds03.scs.illinois.edu/api/results HTTP/1.1
10:58:52:E :OUT1504:Failed response: EOF
```
expand into:
`Failed responce: Connection Failed` or whatever EOF is actually abreviating
At the moment `EOF` is kinda cryptic.

The reason I am targeting more advanced users is because "normal" users will simply come to forum or discord and ask: "My fah is not working" regardless on how detailed the error message is. So if we, advanced users have some sort of hint on what the error means, we could guide others to do some troubleshooting and maybe fixing the issue

**Comment by jcoffland** [comment]
EOF = End of File

**Comment by dymil** [comment]
See, that's totally incomprehensible, and I teach computer networking to undergraduates, and only after you mention it, grok that probably the socket is somewhere wrapped in a file-like interface.

Even though I'm sure this is all output by some HTTPS library under the hood, appending some conjecture to the error message—say, that the underlying TCP socket call returned with '0' or graceful closure—would go a long way. It'd just be a lookup table, basically, to append such text.

**Comment by arisu3** [comment]
I think that "no appropriate assignment" is vague, because it doesn't explain what the issue is. If it said something like "no work currently available for this hardware" then it would make much more sense.

Of course, it would be even better if the error was communicated through the web UI rather than buried in the logs, where the average user will see a bunch of jargon and their eyes will glaze over.

**Comment by muziqaz** [comment]
I agree, but I think for now priority is to work out the bugs and bring the client up to speed with most requested features :)
